### PR TITLE
Schema inference enhancements

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
@h-mayorquin Thanks for reporting the bug

Correcting a typo, enhancing warning info, and debugging exclusion interaction here

### CODE REVIEW

**Comment by h-mayorquin** [comment]
Oh, thanks for taking over the fixing. Let me know when is ready and I can run in the current example to see if it becomes clearer.

Current example for future reference:

```python

/home/heberto/development/neuroconv/src/neuroconv/basedatainterface.py:81: UserWarning: The argument_name 'nwbfile' from the docstring not occur in the method signature, possibly due to a typo.
  return get_json_schema_from_method_signature(self.add_to_nwbfile, exclude=["nwbfile", "metadata"])
/home/heberto/development/neuroconv/src/neuroconv/basedatainterface.py:81: UserWarning: The argument_name 'metadata' from the docstring not occur in the method signature, possibly due to a typo.
  return get_json_schema_from_method_signature(self.add_to_nwbfile, exclude=["nwbfile", "metadata"])
/home/heberto/development/neuroconv/src/neuroconv/basedatainterface.py:81: UserWarning: The argument_name 'The default is False (append mode).' from the docstring not occur in the method signature, possibly due to a typo.
  return get_json_schema_from_method_signature(self.add_to_nwbfile, exclude=["nwbfile", "metadata"])
  ```

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin I am unable to reproduce the `The argument_name 'The default is False (append mode).'` case

Can you give this a try and see how it works?

File: src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py [hunk 1]
```diff
@@ -315,7 +315,6 @@ def add_to_nwbfile(
 
                 metadata['Ecephys']['ElectricalSeries'] = dict(name=my_name, description=my_description)
 
-        The default is False (append mode).
```
Comment (h-mayorquin) [inline src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py:318]: @CodyCBakerPhD it seems that it came from here. Let me test, anyway, this was an error.

**Comment by h-mayorquin** [comment]
After my last changes and yours now I don't get any of those warnings but I saw the tests are failing.

File: src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py:318]: You would be correct!

https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py#L318 would be interpreted as an argument not the description of the argument it is attached to

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin Tests passing now

Might be worth cutting a hotfix release for this

**Comment by h-mayorquin** [comment]
> @h-mayorquin Tests passing now
> 
> Might be worth cutting a hotfix release for this

Yeah, why not, plus I know how to fix images so they display correctly in pipy and I forgot to do it in the latest release.

https://pypi.org/project/ndx-binned-spikes/

File: src/neuroconv/tools/testing/mock_interfaces.py [hunk 2]
```diff
@@ -128,8 +128,7 @@ def __init__(
         self,
         num_channels: int = 4,
         sampling_frequency: float = 30_000.0,
-        # durations: tuple[float] = (1.0,),  # Uncomment when pydantic is integrated for schema validation
-        durations: tuple = (1.0,),
+        durations: tuple[float] = (1.0,),
```
Comment (h-mayorquin) [inline src/neuroconv/tools/testing/mock_interfaces.py:131]: Yei!

File: src/neuroconv/utils/json_schema.py [hunk 3]
```diff
@@ -102,6 +102,9 @@ def get_json_schema_from_method_signature(method: Callable, exclude: Optional[li
     exclude = exclude or []
     exclude += ["self", "cls"]
 
+    split_qualname = method.__qualname__.split(".")[-2:]
+    method_display = ".".join(split_qualname) if "<" not in split_qualname[0] else method.__name__
```
Comment (h-mayorquin) [inline src/neuroconv/utils/json_schema.py:106]: What is the `<` case for?

Comment (CodyCBakerPhD) [reply src/neuroconv/utils/json_schema.py:106]: Sometimes `<locals>` (local namespaces domain) shows up for basic functions. Think it depends on how those functions are defined/imported, but it happens in the tests and I think would happen if you locally defined a data interface in the same script you are trying to run this with

**Review by h-mayorquin — APPROVED** [review]

**Pull request merged**